File: BackEnd/TicketBooking.Domain/Entities/Order.cs
using TicketBooking.Domain.Common;
using TicketBooking.Domain.Enums;

namespace TicketBooking.Domain.Entities;

public class Order : AuditableEntity
{
    public string OrderCode { get; set; } = string.Empty;

    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public decimal TotalAmount { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.PENDING;

    public ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
    public ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
}

File: frontend/ticket-booking-web/src/domain/entities/Order.ts
import type { OrderStatus } from '../enums/OrderStatus';
import type { OrderDetail } from './OrderDetail';
import type { Payment } from './Payment';

export interface Order {
  id: string;
  userId: string;
  totalAmount: number;
  status: OrderStatus;
  createdAt: string;
  updatedAt?: string;
  orderDetails?: OrderDetail[];
  payments?: Payment[];
}
